test/libsolidity/syntaxTests/duplicateFunctions/illegal_names.sol

```diff
@@ -0,0 +1,14 @@
+contract C {
+	function this() public {
+	}
+	function super() public {
+	}
+	function _() public {
+	}
+}
+// ----
+// DeclarationError 3726: (14-41): The name "this" is reserved.
+// DeclarationError 3726: (43-71): The name "super" is reserved.
```

Comment:
One reason why we allowed `this` and `super` is to be able to define interfaces for contracts written in other languages that do not assign this special meaning to these terms. Maybe we can find a middle ground to at least allow external functions with these names?